services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/directory/decorator/DateIndexingDecorator.java

```diff
@@ -0,0 +1,37 @@
+package app.coronawarn.server.services.distribution.assembly.diagnosiskeys.structure.directory.decorator;
+
+import app.coronawarn.server.services.distribution.assembly.diagnosiskeys.structure.directory.DiagnosisKeysDateDirectory;
+import app.coronawarn.server.services.distribution.assembly.structure.directory.decorator.indexing.IndexingDecoratorOnDisk;
+import app.coronawarn.server.services.distribution.assembly.structure.util.ImmutableStack;
+import app.coronawarn.server.services.distribution.config.DistributionServiceConfig;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class DateIndexingDecorator extends IndexingDecoratorOnDisk<LocalDate> {
+
+  private final DistributionServiceConfig distributionServiceConfig;
+
+  public DateIndexingDecorator(DiagnosisKeysDateDirectory directory,
+      DistributionServiceConfig distributionServiceConfig) {
+    super(directory, distributionServiceConfig.getOutputFileName());
+    this.distributionServiceConfig = distributionServiceConfig;
+  }
+
+  /**
+   * Returns the index of the decorated {@link DiagnosisKeysDateDirectory}. By default, the current date (today) will be
+   * excluded from the index. However, if the profile `include-incomplete` is set, the current date will be included.
+   */
+  @Override
+  public Set<LocalDate> getIndex(ImmutableStack<Object> indices) {
+    if (Boolean.FALSE.equals(distributionServiceConfig.getIncludeIncompleteDays())) {
```

Comment:
Consider ``!`` operator for increased clearliness.

Reply:
Had it with a `!` before, but Sonar was (rightfully) complaining about that. Since the boxed type `Boolean` is used, `!distributionServiceConfig.getIncludeIncompleteDays()` might incorrectly pick up a `null` as a false. Using `Boolean.FALSE.equals(...)` was the recommended fix.

Reply:
it would produce a NullPointerException

Reply:
https://rules.sonarsource.com/java/RSPEC-5411